### ISSUE TITLE
Proposal of moving from GitHub Discussions to Discourse

### ISSUE THREAD

**Issue by fharper** [meilisearch/devrel#21]
I will write down a proposal for us to discuss (unless you all agree, haha), but I will suggest we move away from GitHub Discussions to [Discourse](https://www.discourse.org/) so we can have all MeiliSearch discussions at the same place: way easier for someone looking for an answer instead of trying to find in which repo the information is. I'm also thinking when we gonna have a paid product: it won't make sense to have questions about the paid service in GitHub, so we will need an external forum anyway.

**Comment by fharper** [comment]
I'll add more details tomorrow, but in the meantime, here is the little presentation I wanted to use to demonstrate what we have, what need/miss and what we will gain, in this case, with Discourse.

https://docs.google.com/presentation/d/1GWtq3Q0YbMkWh9WGetmW6OshhfR3QN25qxtk1tqj3a4/edit#slide=id.p

**Comment by tpayet** [comment]
What are the key features from Discourse we are missing on GitHub discussion?

**Comment by fharper** [comment]
You can check the small presentation I created, but I'll add more details here tomorrow (was supposed to do it yesterday 😬)

**Comment by fharper** [comment]
So GitHub Discussions are easy to use, and is integrated to GitHub, but it's missing some interesting features that would [make our life easier and would be future-proof](https://www.discourse.org/features).

The access management is tied to the repository, which is something I would prefer not as we can be a lot more liberal with the forum access than the code itself. Which brings me to the fact that one way we will empower the community to contribute outside of just code will be to help with the forum. I've seen it quite successful in the past.

Funny enough, [Discourse](https://www.discourse.org/) is [open source](https://github.com/discourse/discourse): not GitHub discussions, but it's not a real issue since we already use GitHub.

There is a spam management feature which will be useful when we are more popular. Also, for the devrel side, we have a nice dashboard with useful graphs.

Two main aspects for me are game changers. First, the integration with Discourse as really good, so it's easier for us to automate things with [GitHub](https://meta.discourse.org/t/discourse-github/99895), and even more interesting for the developer relations team is the [integration with Orbit](https://www.notion.so/orbitlove/How-to-setup-the-Orbit-Discourse-integration-9b6e9ba978324ecfa8619b79b7e9948d). The second one is at the top of my list: less confusing for the user since it's one link to rule them all. Right now, if someone has a question, they need to find the right repository to write about it. If you have something like forum/discourse/discussions/help.meilisearch.com, you see a page will all sub forums for the integrations, core... It will also give us the ability to add a forum for when we have a paid product. Obviously, we can create a repo for that, but in any cases, I can guarantee that we will have people asking questions in the main repo for the paid product, and it's something I want to separate.

As a bonus, discourse has a gamification aspect, which makes working with a forum more interesting, and can also help us track who helps us and how they helped us to give them some goodies and/or help when we'll have an MVP program.

We can [assign question to people with their plugin](https://meta.discourse.org/t/discourse-assign/58044), which will save us time, and is way better...

So in a nutshell, it doesn't add workload for us as a company (actually, it saves us time), but make it easier for our users/contributors & add a ton of useful features for devrel :D

Any other questions?

**Comment by fharper** [comment]
Just found this in the private repo https://github.com/meilisearch/discuss
![image](https://user-images.githubusercontent.com/1170989/103312375-d5054780-49ea-11eb-8556-1ac2105d47cc.png)

**Comment by tpayet** [comment]
I think we also want to have a place where general discussion and issues around the project should happen instead of Slack. using Discourse, we will have persistent content that people would be able to search through. 

We should also deactivate the discussion feature of the [core repository](https://github.com/meilisearch/meilisearch).

  We also have this [contact page](https://docs.meilisearch.com/resources/contact.html) which is linked on the landing page and could be reworked as a general guide on where to join Meili team and the community depending on the theme of the discussion. Once we have a pretty clear definition of which discussion should happened on the discourse, which discussion on Slack & which discussion on GitHub issues I think we could start working on setting up and structuring Discourse forum!

**Comment by fharper** [comment]
> We also have this [contact page](https://docs.meilisearch.com/resources/contact.html) which is linked on the landing page and could be reworked as a general guide on where to join Meili team and the community depending on the theme of the discussion.

I agree

**Comment by fharper** [comment]
> Once we have a pretty clear definition of which discussion should happened on the discourse, which discussion on Slack & which discussion on GitHub issues I think we could start working on setting up and structuring Discourse forum!

I'll create an issue for that, but IMHO, Slack vs a forum is the choice of people. They both have the same goal, getting helps. Forum having advantages of better history and search function than Slack, but Slack as the advantages of usually quicker reply.

**Comment by fharper** [comment]
blocked by #72

**Comment by tpayet** [comment]
I've just read that [Orbit is planning](https://docs.google.com/document/d/1PFHhZyyYmiIHNufYpS6fX7gUMEJVo3FM0Y0RZa7DuDo/edit) to use [Forem](https://github.com/forem/forem) as their community platform for persistent discussion.

Forem is the platform that powers dev.to. Is this something we might be interested into?

**Comment by fharper** [comment]
I don't think so. We need more like a conventional forum where we can manage the questions, more than a publishing type of forum. I would prefer to stay with Discourse that fit all our needs than trying the new shiny toy :D

**Comment by fharper** [comment]
During the company follow-up meeting, the move from GitHub Discussions to Discourse was confirmed. We will move forward in Q2.

@eskombro suggested we host it ourselves so we can work on an integration and dogfood it!

**Comment by fharper** [comment]
It was decided, I'll create another issue for the move itself

**Comment by ferdi05** [comment]
I would be happy to discuss with the @meilisearch/saas-team @qdequele @laurajwilkinson 
How do we plan to manage support? SHould we create a knowledge base?

**Comment by qdequele** [comment]
I am not convinced by Discourse today about our use of the "forum" media. I would like to get to the limit of the tools we already use before thinking about using other new tools. After that, I won't be the main user of the platform and if it can help the community and the @meilisearch/devrel-team then I'll align myself with your needs.

**Comment by laurajwilkinson** [comment]
I agree with @qdequele that we can optimise our use of the tools we already have before adding others.
I think we have at least two issues to explore here: (1) how do we have conversations with the community, and which tools should we use, and (2) growing our documentation (beyond the current official "Docs") to support self-service help - where users can search for an answer, and where we establish a feedback loop so that frequent questions from the community are turned into content to include in this documentation.
@ferdi05  I suggest starting with [issue 72](https://github.com/meilisearch/devrel/issues/72) to clearly define your comms channels, then we can look at what type of information users ask for that they're not currently able to find in our docs, and decide how to address this.

**Comment by ferdi05** [comment]
Hi @qdequele @laurajwilkinson thanks for your answers. I don't really want to move to Discourse (I'm actually not a big fan), the topic of this issue is misleading and I should have elaborated more.
What I wanted to discuss is the channel(s) that we will use for support and improving docs, knowledge base... I agree that we already have many tools, and we can start by using them more adequately rather than using another one.
@laurajwilkinson I'd be happy to evaluate this in #72!

**Comment by loid345** [comment]
It may be worth considering the nodebb platform as well. It has the same capabilities as discourse. And there is a ready-made integration with meilisearch
https://github.com/oplik0/nodebb-plugin-meilisearch

**Comment by ferdi05** [comment]
thanks @loid345 
At the moment, we're not really thinking of adding Discourse, NodeBB, or anything similar to our communication channels.
There are already many ways to reach us, and we'd rather improve the experience on these existing communication channels rather than create new ones.

**Comment by loid345** [comment]
> thanks @loid345 At the moment, we're not really thinking of adding Discourse, NodeBB, or anything similar to our communication channels. There are already many ways to reach us, and we'd rather improve the experience on these existing communication channels rather than create new ones.

Thanks for the answer. I mean building an online community for meilisearch. No communication channels will replace the usual forum where you can find all the information. This will reduce the burden on the team as community members share their experiences. And get valuable feedback from users. Nodebb is good here because it can run on a very weak server for a few dollars. It costs practically nothing! Without a community, it's hard to develop and promote your product. See how many messages are received per day in the elasticsearch community. You will be very surprised
https://discuss.elastic.co/latest

**Comment by qdequele** [comment]
Hi, now discord also has this kind of forum-like interface. https://discord.com/blog/forum-channels-space-for-organized-conversation
